solidity/test/tbtc/Vault.test.ts

```diff
@@ -0,0 +1,442 @@
+        expect(await bank.balanceOf(vault.address)).to.equal(amount)
+        expect(await bank.balanceOf(account1.address)).to.equal(
+          initialBalance.sub(amount)
+        )
+      })
+
+      it("should mint TBTC", async () => {
+        expect(await tbtc.balanceOf(account1.address)).to.equal(amount)
+        expect(await tbtc.totalSupply()).to.equal(amount)
+      })
+
+      it("should emit Minted event", async () => {
+        await expect(transactions[0])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(3))
+        await expect(transactions[1])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(1))
+        await expect(transactions[2])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(9))
+      })
+    })
+
+    context("when there are multiple minters", async () => {
+      const amount1 = to1e18(13) // 3 + 1 + 9 = 13
+      const amount2 = to1e18(3) // 1 + 2 = 3
+
+      const transactions: ContractTransaction[] = []
+
+      before(async () => {
+        await createSnapshot()
+
+        transactions.push(await vault.connect(account1).mint(to1e18(3)))
+        transactions.push(await vault.connect(account2).mint(to1e18(1)))
+        transactions.push(await vault.connect(account1).mint(to1e18(1)))
+        transactions.push(await vault.connect(account1).mint(to1e18(9)))
+        transactions.push(await vault.connect(account2).mint(to1e18(2)))
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should transfer balances to the vault", async () => {
+        expect(await bank.balanceOf(vault.address)).to.equal(
+          amount1.add(amount2)
+        )
+        expect(await bank.balanceOf(account1.address)).to.equal(
+          initialBalance.sub(amount1)
+        )
+        expect(await bank.balanceOf(account2.address)).to.equal(
+          initialBalance.sub(amount2)
+        )
+      })
+
+      it("should mint TBTC", async () => {
+        expect(await tbtc.balanceOf(account1.address)).to.equal(amount1)
+        expect(await tbtc.balanceOf(account2.address)).to.equal(amount2)
+        expect(await tbtc.totalSupply()).to.equal(amount1.add(amount2))
+      })
+
+      it("should emit Minted event", async () => {
+        await expect(transactions[0])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(3))
+        await expect(transactions[1])
+          .to.emit(vault, "Minted")
+          .withArgs(account2.address, to1e18(1))
+        await expect(transactions[2])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(1))
+        await expect(transactions[3])
+          .to.emit(vault, "Minted")
+          .withArgs(account1.address, to1e18(9))
+        await expect(transactions[4])
+          .to.emit(vault, "Minted")
+          .withArgs(account2.address, to1e18(2))
+      })
+    })
+  })
+
+  describe("redeem", () => {
```

Comment:
Would it make sense to add a case when one redeemer uses multiple vaults to redeem different assets and assert the bank correctly updates its balance?

Reply:
IMO the same situation as with `mint`.